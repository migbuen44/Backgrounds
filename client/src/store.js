import { configureStore } from '@reduxjs/toolkit';
import userLoginReducer from './slices/userLoginSlice';
import loginModalReducer from './slices/loginModalSlice';
import userInfoReducer from './slices/userInfoSlice';

export default configureStore({
  reducer: {
    userLogin: userLoginReducer,
    loginModal: loginModalReducer,
    userInfo: userInfoReducer,
  },
});
