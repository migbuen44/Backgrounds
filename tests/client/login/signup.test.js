import React from 'react';
import ReactDOM from 'react-dom';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer'
import store from '../../../client/src/store';
import Signup from '../../../client/src/components/login/signup';

describe('Signup', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <Provider store={store}>
        <Signup />
      </Provider>, div)
  })

  it('matches snapshot', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <Signup />
      </Provider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  })
})
