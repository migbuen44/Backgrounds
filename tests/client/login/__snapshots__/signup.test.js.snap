// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup matches snapshot 1`] = `
<div>
  <div
    className="lib-login-module-formSectionContainer-2Mzl"
  >
    <h2
      className="lib-login-module-header-3GFZ"
    >
      SignUp
    </h2>
    <form
      className="lib-login-module-userEntryForm-Br2C"
      onSubmit={[Function]}
    >
      <div
        className="lib-login-module-inputContainer-2kgN"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-envelope fa-w-16 lib-login-module-emailIcon-32Yi"
          data-icon="envelope"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M502.3 190.8c3.9-3.1 9.7-.2 9.7 4.7V400c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V195.6c0-5 5.7-7.8 9.7-4.7 22.4 17.4 52.1 39.5 154.1 113.6 21.1 15.4 56.7 47.8 92.2 47.6 35.7.3 72-32.8 92.3-47.6 102-74.1 131.6-96.3 154-113.7zM256 320c23.2.4 56.6-29.2 73.4-41.4 132.7-96.3 142.8-104.7 173.4-128.7 5.8-4.5 9.2-11.5 9.2-18.9v-19c0-26.5-21.5-48-48-48H48C21.5 64 0 85.5 0 112v19c0 7.4 3.4 14.3 9.2 18.9 30.6 23.9 40.7 32.4 173.4 128.7 16.8 12.2 50.2 41.8 73.4 41.4z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
        <input
          className="lib-login-module-email-2xXE"
          onChange={[Function]}
          placeholder="Email"
          type="email"
        />
      </div>
      <div
        className="lib-login-module-inputContainer-2kgN"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-user fa-w-14 lib-login-module-nameIcon-2SIK"
          data-icon="user"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm89.6 32h-16.7c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16h-16.7C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
        <input
          className="lib-login-module-name-2SEE"
          onChange={[Function]}
          placeholder="Name"
          type="text"
        />
      </div>
      <div
        className="lib-login-module-inputContainer-2kgN"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-lock fa-w-14 lib-login-module-passwordIcon-2SgX"
          data-icon="lock"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M400 224h-24v-72C376 68.2 307.8 0 224 0S72 68.2 72 152v72H48c-26.5 0-48 21.5-48 48v192c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V272c0-26.5-21.5-48-48-48zm-104 0H152v-72c0-39.7 32.3-72 72-72s72 32.3 72 72v72z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
        <input
          className="lib-login-module-password-1jMN"
          onChange={[Function]}
          placeholder="Password"
          type="password"
        />
      </div>
      <button
        className="lib-login-module-submitSignUp-iKnm click"
        onClick={[Function]}
        type="submit"
      >
        SignUp
      </button>
    </form>
  </div>
  <div
    className="lib-login-module-toLogin-dwgW click"
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-arrow-left fa-w-14 "
      data-icon="arrow-left"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 448 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M257.5 445.1l-22.2 22.2c-9.4 9.4-24.6 9.4-33.9 0L7 273c-9.4-9.4-9.4-24.6 0-33.9L201.4 44.7c9.4-9.4 24.6-9.4 33.9 0l22.2 22.2c9.5 9.5 9.3 25-.4 34.3L136.6 216H424c13.3 0 24 10.7 24 24v32c0 13.3-10.7 24-24 24H136.6l120.5 114.8c9.8 9.3 10 24.8.4 34.3z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
    <span
      className="lib-login-module-backToLoginMsg-1ssq"
      onClick={[Function]}
    >
      Back to Login
    </span>
  </div>
</div>
`;
